aptoide-views/src/main/java/cm/aptoide/aptoideviews/downloadprogressview/DownloadProgressView.kt

```diff
@@ -0,0 +1,356 @@
+package cm.aptoide.aptoideviews.downloadprogressview
+
+import android.animation.LayoutTransition
+import android.content.Context
+import android.graphics.drawable.Drawable
+import android.os.Build
+import android.util.AttributeSet
+import android.util.Log
+import android.view.View
+import android.widget.FrameLayout
+import androidx.annotation.CheckResult
+import androidx.annotation.VisibleForTesting
+import cm.aptoide.aptoideviews.R
+import cm.aptoide.aptoideviews.common.Debouncer
+import com.tinder.StateMachine
+import kotlinx.android.synthetic.main.download_progress_view.view.*
+import rx.Observable
+import kotlin.math.max
+import kotlin.math.min
+
+
+/**
+ * This view is responsible for handling the display of download progress.
+ *
+ */
+class DownloadProgressView : FrameLayout {
+  private var isPausable: Boolean = true
+  private var payload: Any? = null
+  private var eventListener: DownloadEventListener? = null
+
+  private var debouncer = Debouncer(750)
+
+  private var currentProgress: Int = 0
+
+  private var animationsEnabled = false
+
+  private var stateMachine = StateMachine.create<State, Event, Any> {
+    initialState(State.Queue)
+    state<State.Queue> {
+      onEnter {
+        Log.d("DownloadProgressView", "State.Queue")
+        resetProgress()
+        progressBar.isIndeterminate = true
+        cancelButton.visibility = View.INVISIBLE
+        resumePauseButton.visibility = View.GONE
+        downloadProgressNumber.visibility = View.VISIBLE
+        downloadState.setText(R.string.appview_short_downloading)
```

Comment:
This text is not right in the following cases:

- Updates (should state Updating when in progress and Update paused when paused)
- Downloads (should state Download paused (this is already wrong in production)

ref: https://app.zeplin.io/project/58a5dde3079da73c80cc892e/screen/5aa004cac3e62af32895c266

Reply:
Fixed